pkg/webui/console/components/device-data-form/index.js

```diff
@@ -36,44 +36,82 @@ import PropTypes from '../../../lib/prop-types'
 import m from './messages'
 import validationSchema from './validation-schema'
 
-@bind
 class DeviceDataForm extends Component {
   constructor(props) {
     super(props)
 
-    const { initialValues } = this.props
+    const {
+      initialValues: { supports_join, ids, root_keys = {} },
+    } = this.props
+
+    const external_js = !Boolean(Object.keys(root_keys).length)
     let otaa = true
-    if (initialValues.ids) {
-      otaa = Boolean(initialValues.supports_join)
+    if (ids) {
+      otaa = Boolean(supports_join)
     }
 
     this.state = {
       otaa,
       resets_join_nonces: false,
       resets_f_cnt: false,
+      external_js,
     }
+    this.formRef = React.createRef()
   }
 
+  @bind
   handleOTAASelect() {
     this.setState({ otaa: true })
   }
 
+  @bind
   handleABPSelect() {
     this.setState({ otaa: false })
   }
 
+  @bind
   handleResetsJoinNoncesChange(evt) {
     this.setState({ resets_join_nonces: evt.target.checked })
   }
 
+  @bind
   handleResetsFrameCountersChange(evt) {
     this.setState({ resets_f_cnt: evt.target.checked })
   }
 
+  @bind
+  async handleExternalJoinServerChange(evt) {
+    const external_js = evt.target.checked
+    await this.setState(({ resets_join_nonces }) => ({
+      external_js,
+      resets_join_nonces: external_js ? false : resets_join_nonces,
+    }))
+
+    if (this.formRef && this.formRef.current) {
+      const { setValues, state } = this.formRef.current
+
+      // Reset Join Server related entries if the device is provisined by
+      // an external JS.
+      if (external_js) {
```

Comment:
`join_server_address` should likewise be set back to its default value on uncheck (via `initialValues`).